### PR TITLE
Audit and add/change typescript prop defs in VictoryTooltip

### PR DESCRIPTION
**Pull request by maddles**
Audited VictoryTooltip DefinitelyTyped prop types:

- Updated types for: centerOffset, dx, dy, events
- Added pointerOrientation
- Extended VictoryCommonProps and VictoryLabelableProps to remove: theme, width, groupComponent, height, horizontal, labelComponent

### CODE REVIEW

File: packages/victory-tooltip/src/index.d.ts [hunk 1]
```diff
@@ -4,43 +4,41 @@ import {
...
   StringOrNumberOrCallback,
   VictoryCommonProps,
+  VictoryLabelableProps,
   VictoryNumberCallback,
   VictoryThemeDefinition,
   VictoryStyleObject
 } from "victory-core";
 
-export interface VictoryTooltipProps {
+export interface VictoryTooltipProps extends VictoryCommonProps, VictoryLabelableProps {
```
Comment (boygirl) [inline packages/victory-tooltip/src/index.d.ts:13]: I don't think it's appropriate for VictoryTooltip to extend from VictoryCommonProps. There's a lot it won't use. I'd prefer keeping the few individual props that it does have in common (i.e. `width`, `height` etc) defined on this interface

File: packages/victory-tooltip/src/index.d.ts [hunk 2]
```diff
@@ -4,42 +4,44 @@ import {
...
   index?: number | string;
-  labelComponent?: React.ReactElement;
   orientation?: OrientationTypes | VictoryNumberCallback;
   pointerLength?: NumberOrCallback;
+  pointerOrientation?: OrientationTypes | ((...args: any[]) => OrientationTypes);
   pointerWidth?: NumberOrCallback;
   renderInPortal?: boolean;
   style?: React.CSSProperties;
   text?: StringOrNumberOrCallback | string[] | number[];
-  theme?: VictoryThemeDefinition;
```
Comment (boygirl) [inline packages/victory-tooltip/src/index.d.ts:42]: I think `theme` should stay in this interface.

**Comment by boygirl** [comment]
@maddles Can you check this issue along side your work on the tooltip types? https://github.com/FormidableLabs/victory/issues/1540

File: packages/victory-tooltip/src/index.d.ts (hunk 2 above)
Comment (maddles) [reply packages/victory-tooltip/src/index.d.ts:42]: Sometimes the props are inconsistent between the source and the docs... would you like me to add missing props to the docs?

**Comment by maddles** [comment]
> @maddles Can you check this issue along side your work on the tooltip types? #1540

pointerWidth was added, so #1540 should be resolved in the next release

**Pull request merged**